main.nf

```diff
@@ -1,234 +1,66 @@
-    Perform initial FASTQC
-    ======================
-*/
-
-process pre_trim_fastqc {
-
-    tag { dataset_id } 
-    
-    publishDir "${params.directory}/fastqc", mode: 'copy'
-        
-    cpus 8
-    
-    input:
-        set dataset_id, file(forward), file(reverse) from pre_trim_fastqc
-    
-    output:
-        set dataset_id, file("${dataset_id}_1_fastqc.zip"), file("${dataset_id}_1_fastqc.zip"), file("${dataset_id}_2_fastqc.html"), file("${dataset_id}_2_fastqc.html") into pre_trim_multi_qc
-    """
-        fastqc --noextract --threads ${task.cpus} ${forward}
-        fastqc --noextract --threads ${task.cpus} ${reverse}
-    """
-}
-
-/* 
-    ============================
-    Aggregate pre-FASTQC results
-    ============================
-*/
-
-process pre_trim_multi_qc_run {
-
-    publishDir "${params.out}/report", mode: 'copy'
-
-    input:
-        file(dataset_id) from pre_trim_multi_qc.toSortedList()
-    output:
-        file("multiqc_report_pre.html")
-    """
-        multiqc --filename multiqc_report_pre.html ${params.directory}/fastqc
-    """
-
-}
-
-/* 
-    =======================
-    Generate an MD5sum hash
-    =======================
+#!/usr/bin/env nextflow 
+/*
+    Andersen Lab C. elegans Trimming Pipeline
+    Authors:
+    - Dan Lu <dan.lu@northwestern.edu>
 */
 
-process md5sum_pre {
+params.fastq_path="/projects/b1059/data/fastq/WI/dna/raw"
+params.trimmed_path="/projects/b1059/data/fastq/WI/dna/processed"
 
-    tag { dataset_id } 
+params.fastq_folder="test_data/fastq"
+params.trimmed_folder="${params.fastq_folder}_fastp"
 
-    input:
-        set dataset_id, file(forward), file(reverse) from md5_fastqc
-    output:
-        file('md5.txt') into md5_set
-    """
-    # Command to determine md5
-    
-    __rvm_md5_for()
-    {
-      if builtin command -v md5 > /dev/null; then
-        echo \"\$1\" | md5
-      elif builtin command -v md5sum > /dev/null ; then
-        echo \"\$1\" | md5sum | awk '{print \$1}'
-      else
-        rvm_error "Neither md5 nor md5sum were found in the PATH"
-        return 1
-      fi
-
-      return 0
-    }
-
-        __rvm_md5_for ${forward} | awk '{ print \$0 "\\t${dataset_id}\\t${forward}" }' > md5.txt
-        __rvm_md5_for ${reverse} | awk '{ print \$0 "\\t${dataset_id}\\t${reverse}" }' >> md5.txt
-    """
-}
 
-process collect_md5sum {
+fq=Channel.fromFilePairs("${params.fastq_path}/${params.fastq_folder}/*_{1,2}.fq.gz", flat: true)
 
-    publishDir "${params.out}/report", mode: 'copy'
 
-    input:
-        file(md5) from md5_set.collectFile(name: 'md5sum.txt')
-
-    output:
-        file("md5sum.txt")
 
-    """
-        echo "Great!"
-    """
+process fastp_trim {
 
-}
+    conda "fastp"
 
-/* 
-    ================
-    Perform trimming
-    ================
+    publishDir "${params.trimmed_path}/${params.trimmed_folder}", mode: 'copy', pattern: "*.fq.gz"
+/*    publishDir "${params.trimmed_path}/${params.trimmed_folder}", mode: 'copy', pattern: "*_fastp.json"
```

Comment:
Don't leave commented out code hanging around. You can always go back using git.

Reply:
oh no cannot find the commit hash! but I fixed this somewhere.